src/components/TreeView/TreeViewNode.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import TreeViewExpand from './TreeViewExpand';
+import TreeViewIcon from './TreeViewIcon';
+import TreeViewTextAndIconWrapper from './TreeViewTextAndIconWrapper';
+import TreeViewIndents from './TreeViewIndents';
+
+class TreeViewNode extends Component {
+  constructor(props) {
+    super(props);
+
+    // A node can be set to be expanded by default
+    this.state = {
+      expanded:
+        (props.node.hasOwnProperty('state') && props.node.state.expanded) ||
+        false
+    };
+
+    this.toggleExpand = this.toggleExpand.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+  }
+
+  // Collapse the current node if any of its parents is collapsed. This should
+  // only fire for nodes that are level 2 or greater
+  componentWillReceiveProps(nextProps) {
+    if (!nextProps.visible && nextProps.level > 1) {
+      this.setState(() => ({ expanded: false }));
+    }
+  }
+
+  toggleExpand() {
+    this.setState(prevState => ({ expanded: !prevState.expanded }));
```

Comment:
for the POC... i think using component state is fine... but preference long term is to lift state to redux or use the `controlled()` pattern introduced.

Reply:
yes, I was a little confused when I was building this out, but now I think I understand what I need to do.  I'm starting to explore a stateless version now to prepare for the conversion